Fixes a bizzare double-free bug that was detected by libASAN/libUBSAN.
I still haven't figured it out, it's probably something to do with switch
lifetime scoping.

---

diff --git a/OpenSim/Actuators/Thelen2003Muscle.cpp b/OpenSim/Actuators/Thelen2003Muscle.cpp
index eff537959..ee407a8aa 100644
--- a/OpenSim/Actuators/Thelen2003Muscle.cpp
+++ b/OpenSim/Actuators/Thelen2003Muscle.cpp
@@ -349,22 +349,21 @@ void Thelen2003Muscle::computeInitialFiberEquilibrium(SimTK::State& s) const
         OPENSIM_THROW_FRMOBJ(MuscleCannotEquilibrate, x.what());
     }
 
-    switch(result.first) {
-
-    case StatusFromInitMuscleState::Success_Converged:
+    switch (result.first) {
+    case StatusFromInitMuscleState::Success_Converged: {
         setActuation(s, result.second["tendon_force"]);
         setFiberLength(s, result.second["fiber_length"]);
         break;
-
-    case StatusFromInitMuscleState::Warning_FiberAtLowerBound:
+    }
+    case StatusFromInitMuscleState::Warning_FiberAtLowerBound: {
         log_warn("Thelen2003Muscle initialization: '{}' is at its minimum "
                  "fiber length of {}.", getName(),
                 result.second["fiber_length"]);
         setActuation(s, result.second["tendon_force"]);
         setFiberLength(s, result.second["fiber_length"]);
         break;
-
-    case StatusFromInitMuscleState::Failure_MaxIterationsReached:
+    }
+    case StatusFromInitMuscleState::Failure_MaxIterationsReached: {
         // Report internal variables and throw exception.
         std::ostringstream ss;
         ss << "\n  Solution error " << abs(result.second["solution_error"])
@@ -375,6 +374,7 @@ void Thelen2003Muscle::computeInitialFiberEquilibrium(SimTK::State& s) const
         OPENSIM_THROW_FRMOBJ(MuscleCannotEquilibrate, ss.str());
         break;
     }
+    }
 }
 
 void Thelen2003Muscle::calcMuscleLengthInfo(const SimTK::State& s,
